src/loaders/RasterJsonLoader.js

```diff
@@ -75,13 +75,13 @@ async function initLoader(imageData) {
 }
 
 export default class RasterLoader extends JsonLoader {
-  constructor(params) {
-    super(params);
+  constructor(dataSource, params) {
     const { url, options } = params;
     if (!url && options) {
-      this.url = URL.createObjectURL(new Blob([JSON.stringify(options)]));
-      this.options = undefined;
+      // eslint-disable-next-line no-param-reassign
+      dataSource.url = URL.createObjectURL(new Blob([JSON.stringify(options)]));
     }
+    super(dataSource, params);
     this.schema = rasterSchema;
```

Comment:
Perhaps there should be a `RasterSource` that extends `JsonSource`? The idea of the DataSource is to have a "readonly" type of shared behavior for derived loaders, but the constructor here is mutating the datasource.

Reply:
The issue I ran into here is that the source would need access to the `options` then, which data sources do not now.  I guess the `RasterSource` could have a `set` method on it instead of the `JsonSource` having a `set` method.